pkg/controllers/management/kontainerdrivermetadata/data_getter.go

```diff
@@ -105,47 +105,31 @@ func GetK8sVersionInfo(rancherVersion string, rkeSysImages map[string]v3.RKESyst
 		}
 	}
 	for majorVersion, k8sVersion := range maxVersionForMajorK8sVersion {
-		k8sVersionRKESystemImages[k8sVersion] = rkeSysImages[k8sVersion]
-		k8sVersionSvcOptions[k8sVersion] = svcOptions[majorVersion]
-	}
-	return k8sVersionRKESystemImages, k8sVersionSvcOptions
-}
-
-func GetRKEK8sServiceOptionsWindows(k8sVersion string, svcOptionLister v3.RKEK8sServiceOptionLister, svcOptions v3.RKEK8sServiceOptionInterface) (*v3.KubernetesServicesOptions, error) {
-	names := []string{
-		getWindowsName(k8sVersion),
-		getWindowsName(util.GetTagMajorVersion(k8sVersion)),
-	}
-	var k8sSvcOption *v3.KubernetesServicesOptions
-	for _, name := range names {
-		obj, err := svcOptionLister.Get(namespace.GlobalNamespace, name)
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				return k8sSvcOption, err
-			}
+		// detect the svc options existing or not
+		if _, exist := svcOptions[majorVersion]; !exist {
 			continue
 		}
-		if obj.Labels[sendRKELabel] == "false" {
-			logrus.Infof("Windows svcOption false k8sVersion %s", k8sVersion)
-			return k8sSvcOption, nil
-		}
-		return &obj.ServiceOptions, nil
-	}
-	for _, name := range names {
-		obj, err := svcOptions.GetNamespaced(namespace.GlobalNamespace, name, metav1.GetOptions{})
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				return k8sSvcOption, err
+
+		rkeSysImages := rkeSysImages[k8sVersion]
+		switch osType {
+		case Windows:
```

Comment:
for windows we should also add a check for k8sVersion's. we should continue if `majorVersion < v1.14`

Reply:
Yes, I use the existing checking of `svcOptions[majorVersion]` to avoid those unsupported versions.